src/backend/embedding/extract_file/extract_file.py

```diff
@@ -0,0 +1,173 @@
+            table = [row + [""] * (max_columns - len(row)) for row in table]  # Fill missing cells with empty strings
+            col_widths = [max(len(str(item)) for item in col) for col in zip(*table)]
+            row_format = " | ".join(f"{{:<{w}}}" for w in col_widths)
+            formatted_table = "\n".join(row_format.format(*row) for row in table)
+            return formatted_table.strip()
+
+        except Exception as e:
+            logger.error(f"Error formatting table: {e}")
+            return "Error occurred while formatting table."
+
+    def extract_pdf_content(self, customer_guid: str, file_path: str, filename: str):
+        results = []
+        try:
+            with pdfplumber.open(file_path) as pdf:
+                for page_number, page in enumerate(pdf.pages, start=1):
+                    page_elements = []
+
+                    for block in page.extract_words():
+                        page_elements.append({
+                            "type": "text",
+                            "content": block["text"],
+                            "x0": block["x0"],
+                            "y0": block["top"]
+                        })
+
+                    for table in page.extract_tables():
+                        table_text = self.format_table_as_text(table)
+                        bbox = page.find_tables()[0].bbox  # Get bounding box for the first table
+                        page_elements.append({
+                            "type": "table",
+                            "content": table_text,
+                            "x0": bbox[0],
+                            "y0": bbox[1]
+                        })
+
+                        pil_image = page.to_image(resolution=300).original
+                        numpy_image = np.array(pil_image)
+                        ocr_result = self.ocr_reader.readtext(numpy_image)
+                        ocr_text = " ".join([text[1] for text in ocr_result])
+                        page_elements.append({
+                        "type": "image",
+                        "content": f"Data obtained from image: {ocr_text.strip()}",
+                        "x0": 0,  # Assuming full-page images
+                        "y0": 0
+                        })
+                    page_elements.sort(key=lambda e: (e["y0"], e["x0"]))
+                    page_text = " ".join([element["content"] for element in page_elements])
+                    results.append({
+                        "metadata": {"page_number": page_number},
+                        "text": page_text
+                    })
+
+            # Save extracted content to a new file
+            output_file = f"/tmp/{customer_guid}/{filename}.rawcontent"
+            with open(output_file, "w") as raw_content_file:
+                raw_content_file.write(str(results))
+
+            logger.info(f"Extracted content saved to '{output_file}'")
+
+            # Step 4: Upload the raw content back to MinIO
```

Comment:
Move line 157 to 162 into a function inside class Upload file for chunks and call this new function from line number 63